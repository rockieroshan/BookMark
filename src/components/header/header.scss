@import '../../assets/scss/main.scss';

header {
    @include flexCenter;
    box-shadow: 0 0 0.25em rgb(0 0 0 / 15%);
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    background: $backgroundColor;
    z-index: 10;

    h1 {
        font-size: 3em;
        font-weight: 500;
        color: $whiteColor;
        color: $textColor;
    }
}
