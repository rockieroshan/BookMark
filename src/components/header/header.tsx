import { Component, h } from '@stencil/core';

@Component({
    tag: 'app-header',
    styleUrl: 'header.scss',
    shadow: true,
})

export class Header {

    render() {
        return (
            <header>
                <h1>Bookmarks</h1>
            </header>
        );
    }
}
