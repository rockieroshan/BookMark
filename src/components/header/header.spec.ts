import { newSpecPage } from '@stencil/core/testing';
import { Header } from './header';


describe('Header', () => {
    it('should render Header', async () => {

        const page = await newSpecPage({
            components: [Header],
            html: `<app-header></app-header>`,
        });

        expect(page.root).toEqualHtml(`
    <app-header>
        <mock:shadow-root>
            <header>
                <h1>
                    Bookmarks
                </h1>
            </header>
        </mock:shadow-root>
    </app-header>
  `);

        expect(page).toBeTruthy();

        expect(page.root).toMatchSnapshot()
    });

});