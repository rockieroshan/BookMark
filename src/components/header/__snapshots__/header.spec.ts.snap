// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should render Header 1`] = `
<app-header>
  <mock:shadow-root>
    <header>
      <h1>
        Bookmarks
      </h1>
    </header>
  </mock:shadow-root>
</app-header>
`;

exports[`should render Header 1`] = `
<app-header>
  <mock:shadow-root>
    <header>
      <h1>
        Bookmarks
      </h1>
    </header>
  </mock:shadow-root>
</app-header>
`;
